leaderboard/src/lib.rs

```diff
@@ -17,14 +17,31 @@ impl HttpServer for LeaderboardActor {
         let path = &req.path[1..req.path.len()];
         let segments: Vec<&str> = path.trim_end_matches('/').split('/').collect();
 
-        match (req.method.as_ref(), segments.as_slice()) {
+        let result = match (req.method.as_ref(), segments.as_slice()) {
             ("GET", ["leaderboards"]) => get_leaderboards(ctx).await,
             ("POST", ["leaderboards"]) => create_leaderboard(ctx, deser_json(&req.body)?).await,
             ("GET", ["leaderboards", leaderboard_id]) => get_leaderboard(ctx, leaderboard_id).await,
             ("POST", ["leaderboards", leaderboard_id, "scores"]) => {
                 post_score(ctx, leaderboard_id, deser_json(&req.body)?).await
             }
             (_, _) => Ok(HttpResponse::not_found()),
+        };
+
+        match result {
+            Ok(mut res) => {
+                res.header.extend([
+                    (
+                        "Access-Control-Allow-Origin".to_owned(),
+                        vec!["*".to_owned()],
+                    ),
+                    (
+                        "Content-Type".to_owned(),
+                        vec!["application/json".to_owned()],
+                    ),
+                ]);
+                Ok(res)
+            }
+            Err(e) => Err(e),
```

Comment:
I had a thought afterwards, you _could_ do
```rust
res.as_mut().map(|res| {
  res.header.extend(blah)
  res
})
```

Which would let you do it right after the `match`. Unsure if this feels better though and this looks clear

Reply:
updated!